src/components/AssetDetails/AssetDetailsComponent.jsx

```diff
@@ -78,18 +144,92 @@ class AssetDetailsComponent extends Component {
       }
     ];
 
+    const generateCentreArray = () => {
+      const { centreList } = this.props;
+      const centreArray = [];
+
+      centreList.map(opt => (
+        centreArray.push(opt.centre_name)
+      ));
+
+      return centreArray;
+    };
+
+    const locationDisplay = () => {
```

Comment:
This should be it's own component. Remember that a container component can still be used in a presentation component. Maybe create a folder for `OfficeLocations`, or whatever you decide to name it. I.e.
Create a container:
```js
import connect from 'react-redux';
import isEmpty from 'lodash/isEmpty';

import { loadCentres } from '../../_actions/centres.actions';
import { updateAsset } from '../../_actions/asset.actions';

import OfficeLocations from '../../components/OfficeLocations';

const getLocationList = centers => {
  const centresArray = centers.map(center => center.centre_name);
  return centreArray;
};

const mapStateToProps = ({ asset, assets, centres }) => ({
  centreLoading: centres.isLoading,
  assetDetail: asset.assetDetail,
  location: (asset.assetDetail || {}).asset_location || '';,
  locationList: getLocationList(centres.centreList), the default value of centreList in initialState should be []
  updateLoading: assets.updateLoading,
  showFetchLocations: isEmpty(centres.centreList)
});

export default connect(mapStateToProps, {
  loadCentres,
  updateAsset
})(OfficeLocations);
```

Create the component:
```jsx
import React, { Component } from 'react';
import isEmpty from 'lodash/isEmpty';

export default class OfficeLocations extends Component {
  constructor(props) {
    super(props);
    this.state = {
      show: false,
      location: props.location
    }
  }

  componentDidMount() {
    if (this.props.showFetchLocations) {
      this.props.loadCentres();
    }
  }

  handleChange = (event, data) => {
    event.stopPropagation();
		
    this.setState({
      location: data.value
    });
  };

  toggleFormVisibility = () => {
    this.setState({
      show: !this.state.show
    });
  };
	
  updateLocation = () => {
    const { updateAsset, uuid } = this.props;
    const { asset_code, model_number, serial_number } = assetDetail;

    updateAsset(uuid, {
      asset_location: this.state.location,
      model_number,
      asset_code,
      serial_number
    }).then(() => {
      this.toggleFormVisibility();
    });
  };

  render() {
    const { updateLoading, centreLoading, locationList } = this.props;
    const { location, show } = this.state;

     if (!show) {
       return (
         <Table.Row>
           <Table.Cell className="details-headings">Location</Table.Cell>
           <Table.Cell>
             {location}
             {verifySuperAdmin() && (
                <Icon
                   name="edit"
		  className="asset-detail__table__icon"
		  onClick={this.toggleFormVisibility}
		/>
	     )}
	</Table.Cell>
      </Table.Row>
    );
  }

  return (
    <Table.Row>
      <Table.Cell className="details-headings">Location</Table.Cell>
      <Table.Cell>
        <Form loading={updateLoading}>
	  {/* Preferable, when the data is being updated, show the tiny loading spinner at the end.
	    This will let the user know that the data is being saved.
	  */}
	  {centreLoading && (
	    <div>
	      {location}
	      <LoadingSpinner />
	    </div>
	  )}

	{!centreLoading && (
	  <DropdownComponent
	    customClass="form-dropdown asset-detail__table__dropdown"
	    label="Location"
	    options={generateDropdownOptions(locationList)}
	    placeholder="Select Asset Location"
	    name="assetLocation"
	    value={location}
	    onChange={this.handleChange}
	    upward={false}
	/>
      )}

       <Icon
          name="close"
	  className="asset-detail__table__icon"
	  onClick={this.toggleFormVisibility}
        />
       <Icon
         name="save"
         className="asset-detail__table__icon"
         onClick={this.updateLocation}
       />
      </Form>
    </Table.Cell>
   </Table.Row>
   );
  }
}
```

In `AssetDetailComponent`, use this component instead of what you currently have. I.e.
```jsx
<OfficeLocations uuid={this.props.match.id} />
```

Note that the `updateAssetSuccess` in `asset.actions.js` should update the location in assetDetails.
Therefore, you don't need to call `AddAsset` in `AssetDetailComponent` as you are currently doing.